### ISSUE TITLE
Cache-Control header is broken for html taken from redis cache

### ISSUE THREAD

**Issue by unnamedfeeling** [W3EDGE/w3-total-cache#432]
Steps to reproduce the issue:

1. setup non-file cache for page caching (Redis in my case) and set some bigger caching time (900 seconds)
2. set up some meaningful cache time in Browser Cache -> HTM & XML (I set up 30 seconds)
3. clear all caches
4. load page in incognito mode (Chrome in my case, but any will do)
5. at first load Cache-Control will be set to max-age=30, public and that is fine
6. after cache expiration it will be max-age=0, public

The root of the problem is in PgCache_ContentGrabber.php:1660 - https://github.com/W3EDGE/w3-total-cache/blob/02db48b6de2b45b7b18f4835578535d17b7e231b/PgCache_ContentGrabber.php#L1660 
It checks if cache has expired and if it is just returns 0 instead of lifetime from settings.
A small fix will do here (this code starts from line 1656 of the file):
`
		$bc_lifetime = $this->_config->get_integer(
			'browsercache.html.lifetime' );

		$expires = ( is_null( $time ) ? $curr_time : $time ) + $bc_lifetime;
		$max_age = ( $expires > $curr_time ? $expires - $curr_time : $bc_lifetime );
`

This fixes my issue. Could you please review this one and add it to next release?

**Comment by unnamedfeeling** [comment]
No, the proposed solution will not fix the issue. Trying to find a way to fix this.

**Comment by unnamedfeeling** [comment]
So after playing with this I found out that it works correctly only when page cache time and browser cache are the same. 
But still this seems inconsistent for me - why does browser ignore the max-age=0 (and cloudfront does ignore it too...)?

**Comment by mavas84** [comment]
Hello @unnamedfeeling 

Thank you for reaching out.
The devs are looking into this and we'll get back to you as soon as we have more information.
Thanks!
